Source/Spaceship/OrdinaryBullet.cpp

```diff
@@ -0,0 +1,90 @@
+#include "OrdinaryBullet.h"
+
+const float PI_F = 3.14159265358979f;
+
+OrdinaryBullet::OrdinaryBullet()
+	: _speedValue(200.0f)
+	, _ordinaryBulletSprite()
+	, _bulletTexture()
+	, _textureScale(1.0f, 1.0f)
+{
+	_zOrder = 4;
+	_life = false;
+}
+
+OrdinaryBullet::~OrdinaryBullet()
+{
+	//unsub event
+}
+
+void OrdinaryBullet::Init(sf::Vector2f position, const  sf::Vector2f bulletDirection, sf::Texture& bulletTexture)
+{
+	Add();
+	
+	_textureScale = sf::Vector2f(0.5f, 0.5f);
+	_life = true;
+	_bulletTexture = &bulletTexture;
+	_ordinaryBulletSprite.setTexture(*_bulletTexture);
+	_ordinaryBulletSprite.setScale(_textureScale);
+	_ordinaryBulletSprite.setOrigin(sf::Vector2f(_bulletTexture->getSize().x / 2.0f, _bulletTexture->getSize().y / 2.0f));
+	float angle = std::atan(bulletDirection.x / -bulletDirection.y) * 180.0f / PI_F;	
+	if (bulletDirection.y > 0)
+		angle += 180.0f;
+	
+	_ordinaryBulletSprite.setRotation(angle);
+	SetSpeed(bulletDirection * _speedValue);
+	SetCoordinates(sf::Vector2f(position.x + bulletDirection.x * _bulletTexture->getSize().y / 2.0f, position.y + bulletDirection.y * _bulletTexture->getSize().y / 2.0f));
```

Comment:
You should not depend on 'texture' or some another things to calculate coordinates. what if tomorrow use start to use some another representation of bullet, for example triangle mesh?